lib/ecto/uuid.ex

```diff
@@ -26,6 +44,51 @@ defmodule Ecto.UUID do
     end
   end
 
+  @compile {:inline, c: 1}
+
+  defp c(?0), do: ?0
+  defp c(?1), do: ?1
+  defp c(?2), do: ?2
+  defp c(?3), do: ?3
+  defp c(?4), do: ?4
+  defp c(?5), do: ?5
+  defp c(?6), do: ?6
+  defp c(?7), do: ?7
+  defp c(?8), do: ?8
+  defp c(?9), do: ?9
+  defp c(?A), do: ?a
+  defp c(?B), do: ?b
+  defp c(?C), do: ?c
+  defp c(?D), do: ?d
+  defp c(?E), do: ?e
+  defp c(?F), do: ?f
+  defp c(?a), do: ?a
+  defp c(?b), do: ?b
+  defp c(?c), do: ?c
+  defp c(?d), do: ?d
+  defp c(?e), do: ?e
+  defp c(?f), do: ?f
+  defp c(_),  do: throw(:error)
```

Comment:
You can collapse all those `c/1`'s into:
```elixir
defp c(v) when (v >= ?0 and v<=?9) or (v >= ?a and v<= ?f), do: c
defp c(v) when v >= ?A and v <= ?9), do: v + (?a - ?A) #  Do you really want to allow uppercase even if normalized?
defp c(_), do: throw(:error) # Should probably be an exception
```
It might also be faster in matching, maybe (benchmark if anyone really cares?).